src/commonMain/kotlin/app/cash/turbine/Turbine.kt

```diff
@@ -0,0 +1,201 @@
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.Channel.Factory.UNLIMITED
+import kotlinx.coroutines.channels.ChannelResult
+
+internal const val debug = false
+
+/**
+ * A standalone [Turbine] suitable for usage in fakes or other external test components.
+ */
+public interface Turbine<T>: ReceiveTurbine<T>  {
+  /**
+   * Returns the underlying [Channel].
+   */
+  public override fun asChannel(): Channel<T>
+
+  public suspend fun cancel(cause: Throwable?)
+
+  /**
+   * Add an item to the underlying [Channel] without blocking.
+   *
+   * This method is equivalent to:
+   *
+   * ```
+   * if (!asChannel().trySend(item).isSuccess) error()
+   * ```
+   */
+  public fun add(item: T)
+
+  /**
+   * Assert that the next event received was non-null and return it.
+   * This function will not suspend, and will throw if invoked in a suspending context.
+   *
+   * @throws AssertionError if the next event was completion or an error.
+   */
+  public fun takeEvent(): Event<T>
+
+  /**
+   * Assert that the next event received was an item and return it.
+   * This function will not suspend, and will throw if invoked in a suspending context.
+   *
+   * @throws AssertionError if the next event was completion or an error, or no event.
+   */
+  public fun takeItem(): T
+
+  /**
+   * Assert that the next event received is [Event.Complete].
+   * This function will not suspend, and will throw if invoked in a suspending context.
+   *
+   * @throws AssertionError if the next event was completion or an error.
+   */
+  public fun takeComplete()
+
+  /**
+   * Assert that the next event received is [Event.Error], and return the error.
+   * This function will not suspend, and will throw if invoked in a suspending context.
+   *
+   * @throws AssertionError if the next event was completion or an error.
+   */
+  public fun takeError(): Throwable
+}
+
+public operator fun <T> Turbine<T>.plusAssign(value: T) { add(value) }
+
+/**
+ * Construct a standalone [Turbine] with an [UNLIMITED] buffer size.
+ */
+public fun <T> Turbine(): Turbine<T> = TurbineImpl()
+
+internal class TurbineImpl<T>(
+  private val channel: Channel<T> = Channel(UNLIMITED),
+  private val job: Job? = null,
+) : Turbine<T> {
+  override fun asChannel(): Channel<T> = object : Channel<T> by channel {
+    override fun tryReceive(): ChannelResult<T> {
+      val result = channel.tryReceive()
+      val event = result.toEvent()
+      if (event is Event.Error || event is Event.Item) _ignoreRemainingEvents = true
+
+      return result
+    }
+
+    override suspend fun receive(): T = try {
+      channel.receive()
+    } catch (e: Throwable) {
+      _ignoreRemainingEvents = true
+      throw e
+    }
+  }
+
+  override fun add(item: T) {
+    if (!asChannel().trySend(item).isSuccess) throw IllegalStateException("Added when closed")
```

Comment:
This one doesn't need to go through `asChannel()`, right?

Reply:
Nope, but per above feedback, I'll be removing access to the unwrapped channel anyway